### ISSUE TITLE
Avoid using atomic primitives in the hot path

### ISSUE THREAD

**Issue by lpereira** [microsoft/ntttcp-for-linux#40]
For instance, in `tcpstream.c` and `udpstream.c`, `__sync_fetch_and_add()` is used to update the number of bytes transferred.  This can cause contention between cores if they share cache lines (i.e. false sharing).  Ideally, each thread would update their own counter without any kind of locking, and these numbers would be tallied up after joining the worker threads.

**Comment by simonxiaoss** [comment]
Yes, but how does the main thread to collect those counters from the child thread counters?
